libraries/api/discussion_helper.cpp

```diff
@@ -196,15 +222,56 @@ namespace golos { namespace api {
             auto vshares = db.calculate_vshares(d.net_rshares.value > 0 ? d.net_rshares.value : 0);
 
             u256 r2 = to256(vshares); //to256(abs_net_rshares);
-            r2 *= pot.amount.value;
+            r2 = (r2 * d.reward_weight) / STEEMIT_100_PERCENT;
+            //r2 *= pot.amount.value;
+            r2 *= 16152;
             r2 /= total_r2;
 
+            uint64_t payout = static_cast<uint64_t>(r2);
+
+            payout = std::min(payout, uint64_t(d.max_accepted_payout.amount.value));
+
+            uint128_t reward_tokens = uint128_t(payout);
+
+            share_type author_tokens = reward_tokens.to_uint64();
+
+            share_type curation_tokens = 0;
+            if (d.total_vote_weight > 0 && d.allow_curation_rewards) {
```

Comment:
if `d.allow_curation_rewards` is `false`, it is in any case subtracted from author_tokens.
Check function pay_curators.